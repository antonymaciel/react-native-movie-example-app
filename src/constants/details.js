export const DETAILS_SUCCESS = 'DETAILS_SUCCESS';
export const DETAILS_FAILD = 'DETAILS_FAILD';
export const DETAILS_LOADING = 'DETAILS_LOADING';