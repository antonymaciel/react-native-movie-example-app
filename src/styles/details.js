import { StyleSheet, Dimensions, Platform } from 'react-native';
import generalStyles from './index'
const {height, width} = Dimensions.get('window');
const posterWidth = height / (2.5 * 1.5 * 1.5);
const backdropImageHeight = height / 2.5;
const posterHeight = backdropImageHeight / 1.5;
const posterDownRange = posterHeight / 2;

const styles = StyleSheet.create({
    headerStyle: { 
        borderBottomWidth: 0
    },
    backdropImage: {
        height: backdropImageHeight,
        width
    },
    posterImage: {
        height: posterHeight,
        width: posterWidth,
        borderRadius: 4
    },
    posterImageContainer: {
        position: 'absolute',
        bottom: -posterDownRange,
        marginLeft: 20,
        zIndex: 200,
        ...generalStyles.shadow
    },
    title: {
        position: 'absolute',
        right: 20,
        marginBottom: 23,
        bottom: 0,
        left: posterWidth + 40,
        color: 'white',
        fontSize: 20,
        fontWeight: 'bold',
        textTransform: 'uppercase'
    },
    generalInfo: {
        left: posterWidth + 40,
        width: width - posterWidth - 50,
        marginTop: 23,
        minHeight: posterDownRange - 23
    },
    descriptionAndButtonContainer: {
        flex: 1, 
        margin: 23
    },
    descriptionText: {
        fontSize: 18,
        fontWeight: '500',
        color: '#696969'
    },
    ratingContainer: {
        flexDirection: 'row',
        justifyContent: 'flex-start'
    },
    votesCount: {
        fontWeight: 'bold'
    },
    infoText: {
        fontWeight: '400',
        color: 'black'
    },
    gradient: {
        flex: 1,
        zIndex: 100,
        position:'absolute'
    },
    averageContainer: {
        flexDirection: 'row',
        justifyContent: 'flex-start',
        marginRight: 10
    },
    averageText: {
        color: 'rgb(210, 54, 100)',
        fontWeight: 'bold',
        fontSize: 20
    },
    averageDecimal: {
        color: 'rgb(210, 54, 100)',
        fontSize: 15
    },
    buttonGradient: {
        flex: 1,
        position: 'absolute',
        paddingHorizontal: 25,
        paddingVertical: 10,
        borderRadius: 50,
        width: 200,
        height: 50,
        alignItems: 'center',
        justifyContent:'center',
    },
    buttonContainer: {
        flex: 1,
        alignItems: 'center',
        marginTop: 50
    },
    buttonLabel: {
        color: 'white',
        fontSize: 20,
        fontWeight: 'bold'
    },
    starsContainer: {
        flexDirection: 'row'
    }
});


export default styles;
