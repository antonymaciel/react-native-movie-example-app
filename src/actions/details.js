import Api from "../api/apiMovies"
import { DETAILS_SUCCESS, DETAILS_FAILD, DETAILS_LOADING, DETAILS_FINISH } from '../constants/details'

const movieDetailsSuccess = (movieDetails) => ({
    type: DETAILS_SUCCESS,
    movieDetails
});

const movieDetailsFaild = () => ({
    type: DETAILS_FAILD, 
});

const movieDetailsLoading = () => ({
    type: DETAILS_LOADING,
});

export const movieDetailsFinish = () => ({
    type: DETAILS_FINISH,
});

export const getMovieDetails = (idMovie) => (dispatch) => {
    dispatch(movieDetailsLoading());
    Api.getMovieDetails(idMovie)
    .then((json) => {
        dispatch(movieDetailsSuccess(json));
    })
    .catch(err => {
        console.log(err);
        dispatch(movieDetailsFaild());
    })
  };