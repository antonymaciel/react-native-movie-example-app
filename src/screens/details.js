import React from "react";
import { View, Button, Image, Text} from "react-native";
import { Rating } from 'react-native-elements';
import styles from "../styles/details";
import appStyles from "../styles";
import { IMAGES_BASE_URL } from "../constants/config"

class Details extends React.Component {

    static defaultProps= {
        details: {
            movieDetails: {title: 'Hello',
            genres: [
                {name: 'asdfasd'}, {name:'asdfaerr'}
            ], vote_average: 3, vote_count: 5, overview: "tofoais djfasdto foaisdjfas dtofoais djfasdtof oaisdjfasd tofoaisdjfasd fasd fas df asdf as dfas df asdf asd fas dfasd asd f asd fa sdf as dfa"
        }
        }
    }

    constructor(props){
        super(props);
    }

    render() {
        const { movieDetails } = this.props.details;
        if (movieDetails) {
            const urlbackdropImage = IMAGES_BASE_URL + movieDetails.backdrop_path;
            const urlposterImage = IMAGES_BASE_URL + movieDetails.poster_path;
            let generes;
            movieDetails.genres.forEach((genere) => {
                generes = generes ? generes + ", " + genere.name : genere.name;
            })
            return (
                <View style={appStyles.pageContainer}>
                    <View>
                        <Image style={styles.backdropImage} source={{uri: urlbackdropImage}}/>
                        <Image style={styles.posterImage} resizeMode='cover' source={{uri: urlposterImage}}/>
                        <Text style={styles.title}>{movieDetails.title}</Text>
                    </View>
                    <View style={styles.generalInfo}>
    
                            <Text style={styles.votesCount}>{movieDetails.vote_count} <Text style={styles.infoText}>People watching</Text></Text>
                            <Text style={styles.infoText} numberOfLines={3}>{generes}</Text>
                            <View style={styles.ratingContainer}>
                                <Text>{movieDetails.vote_average}</Text>
                                <Rating
                                    imageSize={20}
                                    readonly
                                    ratingCount={5}
                                    fractions={1}
                                    startingValue={movieDetails.vote_average / 2}
                                    count={1}
                                    ratingColor='red'
                                    //ratingBackgroundColor='gray'
                                    //type='custom'
                                    //ratingTextColor='red'
                                />
                            </View>
                    </View>
                    <View style={styles.descriptionAndButtonContainer}>
                        <Text style={styles.descriptionText} numberOfLines={5}>{movieDetails.overview}</Text>
                        <Button onPress={() => this.props.goToReviews()} title={'Show Reviews'}/>
                    </View>
                </View>
            );
        }
        return <Text>Loading...</Text>;
    }
    
  }

export default Details;
