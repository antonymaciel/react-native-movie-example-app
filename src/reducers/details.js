import { DETAILS_SUCCESS, DETAILS_FAILD, DETAILS_LOADING, DETAILS_FINISH } from '../constants/details'

export const initialState = {
    movieDetails: null,
    loading: false,
    error: false
}


const details = (state = initialState, action) => {
    switch(action.type){
        case DETAILS_SUCCESS:

            return {
                movieDetails: action.movieDetails,
                loading: false, 
                error: false
            };
        case DETAILS_LOADING:
            return { 
                ...state, 
                loading: true,
                error: false
            };
        case DETAILS_FAILD:
            return { 
                ...state, 
                loading: false, 
                error: true
            };
        case DETAILS_FINISH:
            return initialState;   
        default:
                return state;
    } 
}

export default details;