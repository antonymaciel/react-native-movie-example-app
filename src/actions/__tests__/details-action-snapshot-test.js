import mockStore from 'redux-mock-store';
import { getMovieDetails } from '../details';
jest.mock('../../api/apiMovies');

it('should handle details success action', async () => {
    const store = mockStore();
    await store.dispatch(getMovieDetails('1'));
    expect(store.getActions()).toMatchSnapshot();
});

