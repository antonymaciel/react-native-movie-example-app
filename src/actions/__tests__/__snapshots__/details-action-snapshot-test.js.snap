// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle details success action 1`] = `
Array [
  Object {
    "type": "DETAILS_LOADING",
  },
  Object {
    "movieDetails": Object {
      "title": "Spiderman",
    },
    "type": "DETAILS_SUCCESS",
  },
]
`;
